src/main/java/dev/thedocruby/resounding/Cache.java

```diff
@@ -380,8 +380,7 @@ private static HashMap<String, RawMaterial> deserializeMaterials(String name, Li
         HashMap<String, RawMaterial> map = new HashMap<>();
         // treat children as if they had solute: [parent,...] & composition: [0,...] (sets default values)
         if (children != null) {
-            for (String key : children.keySet()) {
-                LinkedTreeMap modified = children.get(key);
+            children.forEach((key, modified) -> {
                 String[] solute = Utils.asArray(new String[0], modified.get("solute"));
                 Double[] composition = Utils.asArray(new Double[0], modified.get("composition"));
                 solute = ArrayUtils.addFirst(solute, name);
```

Comment:
This lambda references a non-static variable outside its scope. This will cause uneccessary JIT compilation, and without any clear gain other than minor cleanliness here. Currying could be used, but then the code merely becomes more confusing and less clean.